File: cpp/28.cpp
#include <iostream>
#include <string>

std::string CapsCounter(const std::string &Frase)
{
    int capitalCount = 0;
    int lowerCount = 0;
    char vogais[] = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};

    for (char c : Frase)
    {
        for (char i : vogais)
        {
            if (c == i)
            {
                if (std::isupper(c)) // Count uppercase vowels
                {
                    capitalCount++;
                }
                else if (std::islower(c)) // Count lowercase vowels
                {
                    lowerCount++;
                }
                break; // Exit the inner loop when a vowel is found
            }
        }
    }

    // Create a result string summarizing the counts
    std::string result = "Total uppercase vowels: " + std::to_string(capitalCount) + "\n";
    result += "Total lowercase vowels: " + std::to_string(lowerCount) + "\n";

    return result; // Return the summary
}

int main()
{
    std::string frase;
    std::cout << "Escreva sua frase: \n";
    std::getline(std::cin, frase); // Read the entire input line

    // Call the function and get the result
    std::string summary = CapsCounter(frase);

    // Output the summary
    std::cout << summary;

    return 0;
}

File: Lua/28.lua
local capitalCount = 0
local lowerCount = 0
local vogais = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}

io.write("Escreva sua frase:")
local Frase = io.read()

local function contadorDeVogais(frase)
    local totalVogais = 0
    for i = 1, #frase do
        local char = frase:sub(i, i)
        for _, vogal in ipairs(vogais) do
            if char == vogal then
                totalVogais = totalVogais + 1
                if char:lower() == char then
                    lowerCount = lowerCount + 1
                else
                    capitalCount = capitalCount + 1
                end
            end
        end
    end
    return totalVogais
end

local result = contadorDeVogais(Frase)
print("Total vowels: " .. result)
print("Uppercase vowels: " .. capitalCount)
print("Lowercase vowels: " .. lowerCount)
